src/tools/rbd/action/Status.cc

```diff
@@ -190,7 +192,75 @@ static int do_show_status(librados::IoCtx& io_ctx, const std::string &image_name
     }
 
     if (!image_cache_str.empty()) {
+      JSONFormattable f;
+      JSONParser p;
+      bool success = p.parse(image_cache_str.c_str(), image_cache_str.size());
+      if (!success) {
         std::cout << "Image cache state: " << image_cache_str << std::endl;
+      } else {
+        decode_json_obj(f, &p);
+        /*
+        utime_t update_time;
+        std::istringstream iss_update_time(f["update_time"]);
+        iss_update_time >> update_time;
+        */
+        bool present = (bool)f["present"];
+        bool empty = (bool)f["empty"];
+        bool clean = (bool)f["clean"];
+        std::string cache_type = f["cache_type"];
+        std::string host = f["pwl_host"];
+        std::string path = f["pwl_path"];
+        uint64_t pwl_size;
+        std::istringstream iss_size(f["pwl_size"]);
+        iss_size >> pwl_size;
```

Comment:
This is too verbose and also not type-safe IIRC.  librbd and rbd CLI generally use `json_spirit` for interacting with JSON.  See `src/json_spirit/json_spirit.h` and grep for examples under `src/librbd` and `src/tools/rbd`.